packages/components-native/src/Form/Form.test.tsx

```diff
@@ -300,6 +300,32 @@ describe("Form", () => {
       });
     });
 
+    it("should call onSubmitSuccess after submit promise resolves", async () => {
+      onSubmitMock.mockImplementationOnce(() => {
+        return Promise.resolve({
+          resolvedPromiseData: "passed",
+        });
+      });
+      const { getByLabelText } = render(<FormTest onSubmit={onSubmitMock} />);
+      const saveButton = getByLabelText(saveButtonText);
+
+      const newValue = "New Value";
+      fireEvent.changeText(getByLabelText(testInputTextPlaceholder), newValue);
+      expect(onChangeMock).toHaveBeenCalled();
+
+      fireEvent(getByLabelText(switchLabel), "onValueChange", true);
+      expect(onChangeSwitchMock).toHaveBeenCalled();
+
+      fireEvent.press(saveButton);
+
+      await waitFor(() => {
+        expect(onSubmitMock).toHaveBeenCalled();
+      });
+      expect(onSuccessMock).toHaveBeenCalledWith({
```

Comment:
I still want to look into a way I can spy on react-hook-form useForm to ensure that handleSubmit promise resolve calls onsuccessmock?